File: python_scripts/script_2.py
#!/usr/bin/env python
'''
	python_scripts/script_2.py
	--------------------------------------------------------
	This script is a working file for this tutorial. You
	will need to edit this file to implement reviewer
	comments.

	You may run this script individually via

		$ python python_scripts/script_2.py
'''


def drawW():
	'''Draw the W body'''
	# read in data
	W = pd.read_csv('data/letter_2.csv',
					header = None)
	x, y = list(W[0]), list(W[1])
	x.append(x[0])
	y.append(y[0])

	# plot data
	ax.fill(x, y, color = 'black')
	ax.plot(x, y, color = 'black', linewidth = 2)

# Solution
# def drawW():
# 	'''Draw the W body'''
# 	# read in data
# 	W = pd.read_csv('data/letter_2.csv',
# 					header = None)
# 	x, y = list(W[0]), list(W[1])
# 	x.append(x[0])
# 	y.append(y[0])
#
# 	# plot data
# 	ax.fill(x, y, color = (0, 0.2, 0.4))  # per reviewer comment
# 	ax.plot(x, y, color = 'black', linewidth = 2)


# You will not need to edit the code below
if __name__ == '__main__':
	# We are running this script on its own
	import pandas as pd
	import matplotlib.pyplot as plt
	fig, ax = plt.subplots(1, 1)		
	ax.axis('off')
	drawW()
	plt.show()
else:
	# We are running this script from within main_script.py
	# Thus all the dependencies and globals have already 
	# been imported.
	drawW()

File: R_scripts/script_2.R
#!/usr/bin/env Rscript
#
# R_scripts/script_2.R
# --------------------------------------------------------
# This script is a working file for this tutorial. You
# will need to edit this file to implement reviewer
# comments.
#
# You can run this script "individually" by 
# calling the `test_W` function defined below.
#

library(ggplot2)

draw_W <- function(canvas) {
  # Adds a letter "W" to an existing ggplot2 context
  
  # read in data
  W <- read.csv(file = 'data/letter_2.csv', 
                header = FALSE)
  names(W) <- c('x', 'y')
  W[nrow(W) + 1,] <- W[1,]
  
  # plot data
  canvas + 
    geom_polygon(
      data=W,
      fill='black', 
      color='black', 
      size=1, 
      aes(x=x, y=y))     
}


test_W <- function() {
  # Displays just the U body
  
  canvas <- ggplot(NULL) + 
    theme(axis.title=element_blank(),
          axis.text=element_blank(),
          axis.ticks=element_blank(),
          panel.background=element_blank(),
          panel.grid=element_blank())
  
  draw_W(canvas)  
}

# Solution
# draw_W <- function(canvas) {
#   # Adds a W body to an existing ggplot2 context
# 
#   # read in data
#   W <- read.csv(file = 'data/letter_2.csv',
#                     header = FALSE)
#   names(W) <- c('x', 'y')
#   W[nrow(W) + 1,] <- W[1,]
# 
#   # plot data
#   canvas +
#     geom_polygon(
#       data=W,
#       fill=rgb(0, 0.2, 0.4),  # per reviewer comment
#       color='black',
#       size=1,
#       aes(x=x, y=y))
# }
